### PR TITLE
Added stats page to navigation bar

### PR DESCRIPTION
**Pull request by calebmah**
Added stats page to navigation bar, and navigation bar to stats page

### CODE REVIEW

**Review by BenMQ — CHANGES_REQUESTED** [review]
This is nice! However, there's quite some duplication here (adding a new page requires changes across all the pages).

Is there  a way that we can avoid this duplication?

**Comment by calebmah** [comment]
Yep, I see your point. I'll look into whether this can be placed as a separate html file or maybe in `navigation-loader.js`!

**Review by calebmah — COMMENTED** [review]
I've made a potentially breaking change of building the entire navigation bar within `navigation-loader.js`, which allows every page to have a consistent navigation bar. 

The reason I say it's potentially breaking is because it requires every `.html` file to include `<script src="/js/navigation-loader.js"></script>` in the header, as well as to use `onload="initiateUI();"` in the body. Additionally, each file must have a `<ul id="navigation"></ul>` element. It also requires wrapping whatever functions needed to load each specific page in a 'buildUI()' function, which seems to be the pattern among all the pages so far. 

This change was made to avoid having to manually edit many files whenever a new page is to be added to the navigation bar. Instead, new links can be added to the `addDefaultNavigation()` function in `navigation-loader.js`.  

We can discuss more about whether this is the best solution for this problem, it's just what I came up with for now!

**Review by BenMQ — COMMENTED** [review]
LGTM! 

I'm gonna let the git keeper this week do another review, a few things

1. do communicate with your teammates about this change!
2. there are a few more pages added (community.html etc ), could you update those too?
3. The server-side rendering unguided project is a relevant technique. It would allow us to produce a consistent header for every page! Give it a go if you are interested!

File: src/main/webapp/js/navigation-loader.js
Comment (BenMQ) [inline src/main/webapp/js/navigation-loader.js:99]: log an error?

**Review by BenMQ — APPROVED** [review]
LGTM as long as we also update the other pages

**Comment by calebmah** [comment]
Sure, although the other pages haven't been merged to master yet. I was going to wait until they were merged to update them. Or should I be modifying those files on their respective branches directly?

**Comment by BenMQ** [comment]
> Sure, although the other pages haven't been merged to master yet. I was going to wait until they were merged to update them. Or should I be modifying those files on their respective branches directly?

Oh I didn't realize. That's fine then, we can update later when they are merged

**Comment by calebmah** [comment]
Great, I'll wait for at least one other review before merging this

File: src/main/java/com/google/codeu/data/Datastore.java
Comment (Jay-kew) [inline src/main/java/com/google/codeu/data/Datastore.java:147]: hey Caleb, could you explain the difference between the two and how it changes? I just did a cursory search on `withDefaults()` which creates instance with null values but didn't really get it

Comment (calebmah) [reply src/main/java/com/google/codeu/data/Datastore.java:147]: I think you're right, `withDefaults()` creates a `FetchOptions` object with `null` for all values, including the `limit`. From the documentation, "`limit` is the maximum number of results the query will return", meaning that in the unlikely case that there are more than 1000 messages in our system, only the first 1000 would be returned making this statistic incorrect. It is not explicitly stated, but I'm assuming having `null` as the `limit` would return all the messages with no limit, which is what we want in this case.

**Review by Jay-kew — APPROVED** [review]

File: src/main/java/com/google/codeu/data/Datastore.java
Comment (Jay-kew) [reply src/main/java/com/google/codeu/data/Datastore.java:147]: ah I see, because I had thought null would be like zero or empty so was confused over there, thanks for clarifying!

**Pull request merged**
